src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmailController.php

```diff
@@ -0,0 +1,172 @@
+     * @param EmailLogsFilter $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, EmailLogsFilter $filters)
+    {
+        $emailConfigurationForm = $this->getEmailConfigurationFormHandler()->getForm();
+        $extensionChecker = $this->get('prestashop.core.configuration.php_extension_checker');
+
+        $testEmailSendingForm = $this->createForm(TestEmailSendingType::class);
+
+        $configuration = $this->get('prestashop.adapter.legacy.configuration');
+        $isEmailLogsEnabled = $configuration->get('PS_LOG_EMAILS');
+
+        $presentedEmailLogsGrid = null;
+
+        if ($isEmailLogsEnabled) {
+            $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+            $emailLogsGridFactory = $this->get('prestashop.core.grid.factory.email_logs');
+            $emailLogsGrid = $emailLogsGridFactory->createUsingSearchCriteria($filters);
+            $presentedEmailLogsGrid = $gridPresenter->present($emailLogsGrid);
+        }
+
+        return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/Email/email.html.twig', [
+            'emailConfigurationForm' => $emailConfigurationForm->createView(),
+            'isOpenSslExtensionLoaded' => $extensionChecker->loaded('openssl'),
+            'smtpMailMethod' => MailMethodOption::SMTP,
+            'testEmailSendingForm' => $testEmailSendingForm->createView(),
+            'emailLogsGrid' => $presentedEmailLogsGrid,
+            'isEmailLogsEnabled' => $isEmailLogsEnabled,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * Process email configuration saving
+     *
+     * @DemoRestricted(redirectRoute="admin_email")
+     * @AdminSecurity("is_granted(['update', 'create', 'delete'], request.get('_legacy_controller')~'_')", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processFormAction(Request $request)
+    {
+        $formHandler = $this->getEmailConfigurationFormHandler();
+        $emailConfigurationForm = $formHandler->getForm();
+        $emailConfigurationForm->handleRequest($request);
+
+        if ($emailConfigurationForm->isSubmitted()) {
+            $errors = $formHandler->save($emailConfigurationForm->getData());
+
+            if (empty($errors)) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+            } else {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_email');
+    }
+
+    /**
+     * Processes test email sending
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function processTestEmailSendingAction(Request $request)
+    {
+        if (!$request->isXmlHttpRequest()) {
```

Comment:
We must be able to execute this request, even if we are not in xhr. Can be in ApiController directory. (in GET)